src/renderer/features/operation/sign/model/wallet-connect-sign-model.ts

```diff
@@ -9,14 +10,44 @@ import { ReconnectStep } from '../lib/constants';
 import { walletModel } from '@renderer/entities/wallet';
 import { isConnectedStep, isReconnectingStep, isTopicExists } from '../lib/utils';
 import { signModel } from './sign-model';
+import { SignResponse } from '../lib/types';
+
+type SignParams = {
+  client: Client;
+  payload: any;
+};
 
 const reset = createEvent();
 const reconnectModalShown = createEvent();
 const reconnectStarted = createEvent<InitReconnectParams>();
 const reconnectAborted = createEvent();
 const reconnectDone = createEvent();
+const signingStarted = createEvent<SignParams>();
 
 const $reconnectStep = createStore(ReconnectStep.NOT_STARTED).reset(reset);
+const $isSigningRejected = createStore(false).reset(reset);
+const $signature = createStore('').reset(reset);
+
+const signFx = createEffect(async ({ client, payload }: SignParams): Promise<SignResponse> => {
+  return client.request(payload);
+});
+
+forward({
+  from: signingStarted,
+  to: signFx,
+});
+
+sample({
+  clock: signFx.doneData,
+  fn: ({ signature }) => signature,
+  target: $signature,
```

Comment:
Shouldn't we also set `$isSigningRejected = false`?

Reply:
no, we reset it when close reject status modal